### PR TITLE
Fix truncation of MaxKbps/TargetKbps/InitialDelayInKB/BufferSizeInKB

### PR DESCRIPTION
**Pull request by SofyaBalandina**

### CODE REVIEW

File: samples/sample_encode/src/sample_encode.cpp [hunk 1]
```diff
@@ -1648,6 +1648,42 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
...
+    {
+        mfxU32 maxVal = std::max<mfxU32>({pParams->MaxKbps, pParams->nBitRate, pParams->InitialDelayInKB, pParams->BufferSizeInKB});
+        mfxU32 newMaxKb = maxVal;
+        mfxU32 newBitRateMultiplier = 1;
+        while (newMaxKb > mfxU16Limit)
+        {
+            newMaxKb = maxVal;
+            newBitRateMultiplier += 1;
+            newMaxKb /= newBitRateMultiplier; 
+        }
```
Comment (dmitryermilov) [inline samples/sample_encode/src/sample_encode.cpp:1665]: why not to just:
newBitRateMultiplier = std::ceil(static_cast<float>(newMaxKb) / newBitRateMultiplier) ;

File: samples/sample_encode/src/sample_encode.cpp [hunk 2]
```diff
@@ -1648,6 +1648,42 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
     }
 #endif
 
+    mfxU16 mfxU16Limit = std::numeric_limits<mfxU16>::max();
+    if ((pParams->MaxKbps && pParams->MaxKbps > mfxU16Limit) ||
```
[the remark this answers is not in the document]
Comment (SofyaBalandina) [reply samples/sample_encode/src/sample_encode.cpp]: yes, done

File: samples/sample_encode/src/sample_encode.cpp (hunk 1 above)
Comment (SofyaBalandina) [reply samples/sample_encode/src/sample_encode.cpp]: changed

File: samples/sample_encode/src/sample_encode.cpp [hunk 3]
```diff
@@ -1648,6 +1648,34 @@ mfxStatus ParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams* p
...
 
+    mfxU16 mfxU16Limit = std::numeric_limits<mfxU16>::max();
+    if (pParams->MaxKbps > mfxU16Limit || pParams->nBitRate > mfxU16Limit ||
+        pParams->InitialDelayInKB > mfxU16Limit || pParams->BufferSizeInKB > mfxU16Limit)
+    {
+        mfxU32 maxVal = std::max<mfxU32>({ pParams->MaxKbps,
+                                           pParams->nBitRate,
+                                           pParams->InitialDelayInKB,
+                                           pParams->BufferSizeInKB });
+        pParams->nBitRateMultiplier += (mfxU16)std::ceil(static_cast<double>(maxVal) / mfxU16Limit);
```
Comment (dmitryermilov) [inline samples/sample_encode/src/sample_encode.cpp:1659]: do we really need '+=' ?

Comment (SofyaBalandina) [reply samples/sample_encode/src/sample_encode.cpp:1659]: I think, yes, user can set own nBitRateMultiplier

Comment (SofyaBalandina) [reply samples/sample_encode/src/sample_encode.cpp]: 'yes, we need it' - for smt, but for sample_encode it is not needed and deleted

File: samples/sample_multi_transcode/src/transcode_utils.cpp [hunk 4]
```diff
@@ -2979,6 +2979,34 @@ mfxStatus CmdProcessor::VerifyAndCorrectInputParams(TranscodingSample::sInputPar
...
 
+    mfxU16 mfxU16Limit = std::numeric_limits<mfxU16>::max();
+    if (InputParams.MaxKbps > mfxU16Limit || InputParams.nBitRate > mfxU16Limit ||
+        InputParams.InitialDelayInKB > mfxU16Limit || InputParams.BufferSizeInKB > mfxU16Limit)
+    {
+        mfxU32 maxVal = std::max<mfxU32>({ InputParams.MaxKbps,
+                                           InputParams.nBitRate,
+                                           InputParams.InitialDelayInKB,
+                                           InputParams.BufferSizeInKB });
+        InputParams.nBitRateMultiplier += (mfxU16)std::ceil(static_cast<double>(maxVal) / mfxU16Limit);
```
Comment (dmitryermilov) [inline samples/sample_multi_transcode/src/transcode_utils.cpp:2990]: I assume if user set TargetKbps > std::numeric_limits<mfxU16>::max() and some nBitRateMultiplier > 1 it's already a logical violation. Unzero nBitRateMultiplier means user is aware about TargetKbps  precision limitation so IMO better just to drop user defined value.

Comment (SofyaBalandina) [reply samples/sample_multi_transcode/src/transcode_utils.cpp]: removed

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**